### PR TITLE
MRC-143 taxable profits table introduced.

### PR DESCRIPTION
**Pull request by PavelVjalicin**

### CODE REVIEW

File: app/utils/CurrencyUtils.scala [hunk 1]
```diff
@@ -32,4 +32,8 @@ object CurrencyUtils {
       .format(value)
       .replace("GBP", "£")
       .replace(".00", "")
+
+  def decimalFormat(value: Number): String = currencyFormatter
+    .format(value)
+    .replace("GBP", "£")
```
Comment (girishkamat) [inline app/utils/CurrencyUtils.scala:38]: Can we add a unit test for this?

File: app/views/helpers/FullResultsPageHelper.scala [hunk 2]
```diff
@@ -366,4 +367,91 @@ object FullResultsPageHelper extends ViewHelper {
...
+              ),
+              TableRow(
+                content = HtmlContent(CurrencyUtils.decimalFormat(taxableProfit)),
+                classes = "govuk-table__cell--numeric"
+              )
+            ),
+            Seq(
+              TableRow(content = HtmlContent(p(messages("fullResultsPage.taxableProfit.distributions")))),
+              TableRow(
+                content = HtmlContent(CurrencyUtils.decimalFormat(BigDecimal(distributions) / totalDays * d1.days)),
```
Comment (girishkamat) [inline app/views/helpers/FullResultsPageHelper.scala:405]: why are we computing the adjusted values on the fly, dont we have methods in calculator result to get it?

If they dont exist, we need to add them

Comment (PavelVjalicin) [reply app/views/helpers/FullResultsPageHelper.scala:405]: The values in the table don't come from back-end those are direct form values (taxable income, distributions, both).  Should we add those to backend?

Comment (girishkamat) [reply app/views/helpers/FullResultsPageHelper.scala:405]: As far as I understand, the values you are displaying are adjust profits and distributions for the tax year, which is automatically computed in the calculator result.

Comment (PavelVjalicin) [reply app/views/helpers/FullResultsPageHelper.scala]: Implemented via backend. Intergated with last commit.

File: app/views/helpers/FullResultsPageHelper.scala [hunk 3]
```diff
@@ -366,4 +367,89 @@ object FullResultsPageHelper extends ViewHelper {
...
+        )
+      )
+    }
+
+    calculatorResult.fold(single => HtmlFormat.empty)(dual =>
+      dual.year1 -> dual.year2 match {
+        case (d1: FlatRate, d2: MarginalRate) => table(d1, d2)
+        case (d1: MarginalRate, d2: FlatRate) => table(d1, d2)
+        case _                                => HtmlFormat.empty
+      }
```
Comment (girishkamat) [inline app/views/helpers/FullResultsPageHelper.scala:452]: Isnt there is taxable profit table for Single(marginalrate) and Dual(marginalrate, marginalrate) scenario?

Comment (PavelVjalicin) [reply app/views/helpers/FullResultsPageHelper.scala:452]: No, I think the logic here is right. This table is only displayed in a very specific scenario.

Comment (girishkamat) [reply app/views/helpers/FullResultsPageHelper.scala:452]: Cool, looks good!

**Review by girishkamat — APPROVED** [review]

**Pull request merged**